frontend/src/components/RoleMap/RoleMapForm.tsx

```diff
@@ -0,0 +1,341 @@
+                                        const updatedRoleMap = [...roleMap];
+                                        updatedRoleMap.splice(index, 1);
+                                        setRoleMap(updatedRoleMap);
+                                    }}/>
+                            </div>
+                        }
+                        key={index}
+                    >
+                        {renderRoleDetails(role, "role")}
+                    </Collapse.Panel>
+                ))}
+            </Collapse>
+
+            <Button className={styles.addPermissionButton} type="default" onClick={() => {
+                setRoleMap([...roleMap, {name: "", permit: [], deny: []}]);
+            }}>
+                Add Role
+            </Button>
+        </>
+    )
+
+    const subrolesSection = (
+        <>
+            <h2>Subroles</h2>
+            <Collapse accordion>
+                {subroleMap.map((subrole, index) => (
+                    <Collapse.Panel
+                        header={
+                            <div className={styles.roleHeader}>
+                                <Input
+                                    placeholder="Subrole Name"
+                                    value={subrole.name}
+                                    onChange={(e) => {
+                                        const oldName = subrole.name;
+                                        const newName = e.target.value;
+
+                                        handleSubroleNameChange(index, newName, oldName);
+                                    }}
+                                />
+
+                                <Button
+                                    type="text"
+                                    danger
+                                    icon={<MdCancel style={{fontSize: "120%"}}/>}
+                                    onClick={() => {
+                                        const updatedSubroleMap = [...subroleMap];
+                                        updatedSubroleMap.splice(index, 1);
+                                        setSubroleMap(updatedSubroleMap);
+                                    }}/>
+                            </div>
+                        }
+                        key={index}
+                    >
+                        {renderRoleDetails(subrole, "sub")}
+                    </Collapse.Panel>
+                ))}
+            </Collapse>
+
+            <Button className={styles.addPermissionButton} type="default" onClick={() => {
+                setSubroleMap([...subroleMap, {name: "", permit: [], deny: []}]);
+            }}>
+                Add subrole
+            </Button>
+        </>
+    )
+
+    const onCancel = () => {
+        Modal.confirm({
+            title: 'Are you sure?',
+            content: 'Changes you made will be lost.',
+            okText: 'Yes',
+            cancelText: 'No',
+            onOk: () => {
+                navigate('/Roles');
+            }
```

Comment:
![image](https://github.com/user-attachments/assets/f07beeb8-7324-453b-9c8f-72085e251ded)
To nie jest wycentrowane

Reply:
Nie będę teraz uruchamiał, ale to jest wycentrowane. Z tym że do środka ekranu, nie tylko kontenera na role.

Reply:
Meh imo źle wygląda, ale to tylko moja opinia, może tak zostać